docs/DeveloperGuide.md

```diff
@@ -464,6 +479,165 @@ As the `Model` class was built prior to the implementation of this feature, we d
+
+Note: `deleteExam` follows a similar structure, differing in the arguments parsed and the methods called on the `Model` component (e.g. deleteting from `UniqueExamList` instead of adding to it).
+
+<br>
+
+#### **Select Exam Command** : `selectExam`
+
+The `selectExam` command allows users to select an exam from the ``UniqueExamList`. The selection of exams is heavily used in conjunction with our exam score features.
+
+##### Parsing User Input
+
+The `selectExam` command takes in one argument from the user: the index of the exam to be selected. `SelectExamCommandParser` checks that the user has provided the index and that the index is valid.
+
+##### Executing the Command
+
+On execution, the `SelectExamCommand` retrieves the the selected exam from the `UniqueExamList` based on the user input index. The selected exam is then stored in `Model` under the `selectedExam` field.
+
+If the index is invalid (e.g. the index is greater than the size of the exam), a `CommandException` is thrown.
+
+<br>
+
+#### **Deselect Exam Command** : `deselectExam`
+
+The `deselectExam` command allows users to deselect the currently selected exam.
+
+##### Parsing User Input
+
+The `deselectExam` command does not take any arguments from the user. Hence, a `DeselectExamCommandParser` is not required. `AddressBookParser` directly creates a `DeselectExamCommand` object.
+
+##### Executing the Command
+
+On execution, the `DeselectExamCommand` sets the `selectedExam` in `ModelManger` to null. If there is no exam currently selected, a `CommandException` is thrown.
+
+<br>
+
+#### **Sequence diagrams illustrating exam selection**
+
+The following sequence diagram illustrates the interactions between the Logic and Model when the `SelectExamCommand` is executed.
+
+<puml src="diagrams/SelectExamSequenceDiagram.puml" alt="Sequence Diagram for the parsing of `selectExam` Command" />
+
+Notes:
+- The `ObservableList<Exam>` object is what is returned when retrieving the `UniqueExamList`. This prevents unwanted modifications to the `UniqueExamList` when retrieving the selected exam.
+- `deselectExam` follows a similar structure as the diagram above, differing in the arguments parsed and the methods called on the `Model` component (i.e. calling `deselectExam` on `Model` instead of `selectExam`).
+
+<br>
+
+#### **Considerations for Exam Features**
+
+##### Using a selection system for exams
+
+We decided to implement a selection system for exams to complement the exam score feature. The application would only display the scores of the selected exam, making it easier for users to manage and view the scores.
+
+Our alternative design was to display the scores of all exams at once on every person. However, this alternative design would have made the UI cluttered and less user-friendly. The selection system allows users to focus on the scores of a specific exam, making it easier to view and manage the scores.
```

Comment:
Good elaboration on the alternatives we could have gone for and why we chose to follow our current implementation.